src/iuse.cpp

```diff
@@ -2509,13 +2509,33 @@ int iuse::dig( player *p, item *it, bool t, const tripoint & )
     int moves;
 
     if( g->m.has_flag( "DIGGABLE", pnt ) ) {
-        if( g->m.ter( pnt ) == t_pit_shallow ) {
+        const ter_id ter = g->m.ter( pnt );
+        if( ter == t_pit_shallow ) {
             if( p->crafting_inventory().has_quality( DIG, 2 ) ) {
                 moves = MINUTES( 90 ) / it->get_quality( DIG ) * 100;
             } else {
                 p->add_msg_if_player( _( "You can't deepen this pit without a proper shovel." ) );
                 return 0;
             }
+        } else if( ter == t_grave ) {
+            if( g->u.has_trait_flag( "SPIRITUAL" ) && !g->u.has_trait_flag( "PSYCHOPATH" ) &&
+                g->u.query_yn( _( "Would you really touch the sacred resting place of the dead?" ) ) ) {
+                add_msg( m_info, _( "You are really going against your beliefs here." ) );
+                g->u.add_morale( MORALE_GRAVEDIGGER, -50, -100, 48_hours, 12_hours );
+                if( one_in( 3 ) ) {
+                    g->u.vomit();
+                }
+            } else if( g->u.has_trait_flag( "PSYCHOPATH" ) ) {
+                p->add_msg_if_player( m_good, _( "This is fun!" ) );
```

Comment:
I'd expand the `This` part of the message. Like "Messing with the graves...", "Exhuming graves...", "Ravaging graves..", "Being a graveopener..." or something like that.

Reply:
Applied, to other messages also.